File: JAVA/S2_SORTING/L1_BASIC/selectionsort.c
#include<stdio.h>
void selectionsort(int arr[],int n)
{
    for(int i=0;i<n-1;i++)
    {
        int min=i;
        for(int j=i;j<n;j++)
        {
            if(arr[j]<arr[min])
            {
                min=j;
            }
        }
        if(min!=i)
        {
            int t=arr[min];
            arr[min]=arr[i];
            arr[i]=t;
        }
    }
}
int main()
{
    int n;
    scanf("%d",&n);
    int arr[n];
    for(int i=0;i<n;i++)
    {
        scanf("%d",&arr[i]);
    }
    selectionsort(arr,n);
    for(int i=0;i<n;i++)
    {
        printf("%d ",arr[i]);
    }
    return 0;
}

File: python/S2_SORTING/L1_BASIC/selectionsort.py
class sort:
    def __init__(self,arr):
        self.arr=arr
    def selectionsort(self):
        n=len(self.arr)
        for i in range(n-1):
            min_idx=i
            for j in range(i,n):
                if self.arr[j]<self.arr[min_idx]:
                    min_idx=j
            if i!=min_idx:
                self.arr[i],self.arr[min_idx]=self.arr[min_idx],self.arr[i]
if __name__=="__main__":
    n=int(input("enter size of array :"))
    arr=list(map(int,input("enter element of array :").split()))
    s=sort(arr)
    s.selectionsort()
    for x in arr:
        print(x,end=" ")